### DISCUSSION TITLE
Referencing a flow within a connection event

### DISCUSSION THREAD

**Discussion by malwareinfosec** [mitmproxy/mitmproxy#6710] [General]
I'm using a connection event to get the IP address of the remote server (_data.server.peername_). I would like to mark the corresponding flow in the UI but I don't know how to reference the flow within the _def server_connected_ function since it does not seem to accept flow as a parameter.

![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/f7cde672-d202-4933-87d9-400a3a7455cc)

_**Addon error: fiddleitm.server_connected() missing 1 required positional argument: 'flow'**_

**Comment by malwareinfosec** [comment]
Would it be better to access the remote IP address value from the flow detail view instead? I can't seem to find anything in the documentation to access that property.

![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/9ef56391-f420-4104-9223-0257689dc828)

![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/adadbdc0-36be-46ef-b91e-aad2301c4b9d)

**Comment by mhils** [comment]
The easiest way forward here is probably to inspect `flow.server_conn.peername` in the `responseheaders` or the `response` hook.[^1]

Alternatively you could work with a WeakKeyDict mapping server connection objects to flows, but from what you mentioned that sounds like unnecessary complexity here.

[^1]: You may already see an IP address in `server_conn.peername` in the request hook, but that's not guaranteed for some proxy modes).

**Comment by malwareinfosec** [comment]
Thanks @mhils, this is what I was looking for.

![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/3bc24662-b961-42f3-8177-ed5fa53a5b67)
